### PR TITLE
Feat ts support

### PR DESCRIPTION
**Pull request by babak-zoopit**
### what
- [x] my commit messages describe the changes and comply to [conventionalcommits](https://www.conventionalcommits.org/en/v1.0.0/)
A js and a declaration file added to the module.
semantic release also added

### why

To be able to use it in typescript codes.

### side notes

I prefer this way to leave index completely unchanged.

### CODE REVIEW

**Review by Smembe812 — CHANGES_REQUESTED** [review]
could you remove the `on pull_request event` block because the `on push event` also listens to pull requests

**Review by Smembe812 — CHANGES_REQUESTED** [review]
https://github.com/zoopit/amqp-helper/pull/2/commits/93e975ef90e69ead37b459086a7f9bff0b4d0796

could you try using [util.promisify](https://nodejs.org/api/util.html#utilpromisifyoriginal) instead of Promise((resolve, reject))? Other wise it should be fine how it is now

**Review by Smembe812 — COMMENTED** [review]
> could you remove the `on pull_request event` block because the `on push event` also listens to pull requests

I see you have already removed it in 0d48ec80defc7939355647d2d514dffbaeeda362 @babak-zoopit 👍

**Comment by ilibar-zpt** [comment]
why did you specify branches in CI and semantic release?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '16'
```
Comment (ilibar-zpt) [inline .github/workflows/ci.yml]: any reason to not use 18?

File: package.json [hunk 2]
```diff
@@ -1,19 +1,24 @@
 {
-  "name": "amqp-helper",
+  "name": "@zoopit/amqp-helper",
   "version": "2.0.5",
   "description": "Helper functions for AMQP",
   "main": "src/index.js",
```
Comment (ilibar-zpt) [inline package.json:5]: this one needs a change, and if d.ts has same basename then you don't need `types` key

File: .github/workflows/ci.yml (hunk 1 above)
Comment (babak-zoopit) [reply .github/workflows/ci.yml]: https://nodejs.org/en/

16 is recommended for most users right now.
I can change it.
Let me know if it's better to do it.

File: package.json (hunk 2 above)
Comment (babak-zoopit) [reply package.json:5]: Good information.
I'll check and remove it.

**Comment by babak-zoopit** [comment]
> why did you specify branches in CI and semantic release?

In CI:
To keep sync only master with registry.

In releaserc.json:
I got error without specifying branches (however [here](https://github.com/semantic-release/semantic-release/blob/HEAD/docs/usage/configuration.md#branches) specified that branches has a default value)

**Comment by ilibar-zpt** [comment]
> In CI:
> To keep sync only master with registry.

then how are we supposed to know if CI checks pass on your PR?

> In releaserc.json:

should not change it
what was the error? feel free to ask next time

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ilibar-zpt) [reply .github/workflows/ci.yml]: I'd upgrade now

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+      - run: npm ci
```
Comment (ilibar-zpt) [inline .github/workflows/ci.yml:16]: include `npm audit --audit-level=high` step before `ci`

**Comment by ilibar-zpt** [comment]
consider adding `.editorconfig` for consistency

**Review by Smembe812 — APPROVED** [review]
Could you add some example of how to use it like how you did in https://github.com/zoopit/routing/blob/init-api/api/test/test-helper/amqp.js? Otherwise, it looks fine to me

File: package.json [hunk 4]
```diff
@@ -1,19 +1,22 @@
...
   "scripts": {
+    "release": "semantic-release",
     "test": "mocha"
```
Comment (ilibar-zpt) [inline package.json]: include TS declaration validation to the test

**Review by ilibar-zpt — CHANGES_REQUESTED** [review]
comments to be adressed

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ilibar-zpt) [reply .github/workflows/ci.yml]: any changes? there's old unresolved comment

File: package.json (hunk 2 above)
Comment (ilibar-zpt) [reply package.json:5]: any changes? there's old unresolved comment

File: package.json [hunk 5]
```diff
@@ -1,19 +1,23 @@
...
+    "test": "tsc src/ts-support.d.ts; mocha "
   },
   "author": "trygve@grimnes.org",
   "license": "ISC",
   "dependencies": {
+    "@types/amqplib": "^0.8.2",
     "amqplib": "^0.4.0",
-    "sha1": "^1.1.1",
-    "zlog": "^1.0.0"
+    "mocha": "^10.0.0",
```
Comment (ilibar-zpt) [inline package.json]: why is it a production dependency?

File: package.json [hunk 6]
```diff
@@ -1,19 +1,23 @@
 {
-  "name": "amqp-helper",
+  "name": "@zoopit/amqp-helper",
   "version": "2.0.5",
   "description": "Helper functions for AMQP",
   "main": "src/index.js",
   "scripts": {
-    "test": "mocha"
+    "release": "semantic-release",
+    "test": "tsc src/ts-support.d.ts; mocha "
```
Comment (ilibar-zpt) [inline package.json]: why `;`?

File: src/ts-support.d.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { Connection as Connection1 } from "amqplib" 
+
+// Just declaration file for ts-support.js
+
+
+export function connect(uri: string): Promise<Connection1>;
+
+export function sendWithResponseQueue<T,R>(connection: Connection1 , queueOut: string , queueIn: string , body: T): Promise<R>;
+
+export type Connection = Connection1
```
Comment (ilibar-zpt) [inline src/ts-support.d.ts]:
```suggestion
import type * as AMQP from "amqplib" 

// Just declaration file for ts-support.js


export function connect(uri: string): Promise<AMQP.Connection>;

export function sendWithResponseQueue<T,R>(connection: AMQP.Connection , queueOut: string , queueIn: string , body: T): Promise<R>;

export type Connection = AMQP.Connection
```

**Review by ilibar-zpt — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (babak-zoopit) [reply package.json:5]: had removed in this commit: [e570070](https://github.com/zoopit/amqp-helper/pull/2/commits/e5700708740cc86d80b4a5b776560250a1e09feb)

**Comment by Smembe812** [comment]
Maybe close the PR @babak-zoopit ?

**Comment by ilibar-zpt** [comment]
closed in favor of https://github.com/zoopit/amqp-helper-ts/pull/1

**Pull request closed without merging**